tests: added some more pexpect scripts

haukepetersen opened the pull request:
I was starting task 2 from the release specs (https://github.com/RIOT-OS/Release-Specs/blob/master/02-tests/02-tests.md). Doing so, I found it not only inconvenient but also a waste of time, to do many of these tests manually... I think we all agree, that the goal must be to fully automate these tests!

So this PR is one small step towards this goal: I started to add `pexpect` scripts for tests, where the test result is more or less clear on the test's result state. If everyone agrees on proceeding on this, it would be nice if others could come to the rescue and help improving our current state so we can up this (and all future) release testing.

EDIT
~~NOTE 1: This PR contains also a small fix to allow for simply switching to a different terminal program (as `pyterm` has some trouble with `EOF` characters). This is essential for running the `make test` targets on actual hardware. With this fix, one can simply run e.g. 
`TERMINAL=picocom BOARD=samr21-xpro make all flash test`
This change is taken and adapted from  #7258 -> thanks to @kaspar030!~~ -> moved to its own PR #7758

NOTE 2: I have written a small script, that tries to run `make test` for every application in `tests/`. It gives a quick overview on the tests that actually support this target and their test results -> https://github.com/haukepetersen/riotsandbox/tree/master/tools/tester

kaspar030 commented:
> This PR contains also a small fix to allow for simply switching to a different terminal program

If you factor that out, I'll review it right away.

miri64 commented on tests/bitarithm_timings/tests/01-run.py:
```diff
@@ -0,0 +1,23 @@
+    child.expect_exact("Start.")
+    child.expect('\+ bitarithm_msb: \d+ iterations per second', timeout=20)
+    child.expect('\+ bitarithm_lsb: \d+ iterations per second', timeout=20)
+    child.expect('\+ bitarithm_bits_set: \d+ iterations per second', timeout=20)
```
can you add the `timeout` parameter to `run()` instead, please?

haukepetersen replied:
indeed.

haukepetersen commented:
> If you factor that out, I'll review it right away.

Will do

haukepetersen commented:
done

miri64 commented on tests/buttons/tests/01-run.py:
```diff
@@ -0,0 +1,29 @@
+    #     return
+
+    # child.expect_exact(" -- Try pressing buttons to test.")
+    # child.expect_exact("[SUCCESS]")
```
Please remove comments (also maybe just check `[SUCCESS]` above instead of `-- Available buttons:`)

haukepetersen replied:
seems like I have checked in something incomplete, will fix.

miri64 commented on tests/buttons/tests/01-run.py:
```diff
@@ -0,0 +1,29 @@
+
+def testfunc(child):
+    child.expect_exact("On-board button test")
+    index = child.expect([
```
You don't check `index` anymore, please remove assignment

haukepetersen replied:
now I do use it :-)

haukepetersen commented:
fixed issues.

Also did two more changes:
- rebased on top of #7758
- removed the resetting of the `TERMFLAGS` variable. As far as I can see it, this is not needed anymore, right?

kaspar030 commented:
> removed the resetting of the TERMFLAGS variable. As far as I can see it, this is not needed anymore, right?

+1

miri64 commented on tests/buttons/tests/01-run.py:
```diff
@@ -0,0 +1,26 @@
+        r" -- Available buttons: \d+"
+    ])
+
+    if index == 1:
```
What if the board has more buttons?

miri64 replied:
Sorry ... this isn't the result of the regex, but the result of the branching. So all is correct :-)

haukepetersen replied:
What has `index` to do with the number of buttons?

miri64 replied:
Nothing. I was just confused ;-)

miri64 commented:
Please squash

haukepetersen commented:
rebased. I think it makes sense to also backport this PR, so the tests can be used in RC2 right away...

miri64 commented:
Isn't a backport of #7758 required for that as well?

haukepetersen commented:
yes

miri64 commented:
Backport provided in #7805.
